server/dash/dashboards/neighborhood.py

```diff
@@ -30,59 +32,97 @@
 
 def populate_options():
     values = []
-    for i in df.sort_values('council_name').council_name.unique():
-        values.append(
-            {
-                'label': i,
-                'value': i
-            }
-        )
+    for i in df.sort_values("council_name").council_name.unique():
+        values.append({"label": i, "value": i})
     return values
 
 
 # Define callback to update graph
 @app.callback(
-    Output('graph1', 'figure'),
-    [Input("council_list", "value")]
+    Output("graph1", "figure"),
+    [Input("council_list", "value"), Input("data_type", "value")],
 )
-def update_figure(selected_council):
+def update_figure(selected_council, selected_timeframe):
+    """Creates comparison graph
+
+    Args:
+        selected_council:
+          A string representing the council selected from the drop down menu.
+        selected_timeframe:
+          An integer representing the number of days around which to compute the moving average.
+          This is selected with radio buttons at the bottom of the graph.
+
+    Returns:
+        fig:
+          Plotly graph that compares the selected moving average between the chosen council
+          and the moving average of the avg of all the 99 neighborhood councils.
+    """
+    neighborhood_sum_df = (
+        df[df.council_name == selected_council]
+        .groupby(["created_date"])
+        .agg("sum")
+        .reset_index()
+    )  # noqa
+    total_sum_df = df.groupby(["created_date"]).agg("sum").reset_index()
+
+    total_sum_df["nc_ma"] = (
+        total_sum_df.counts.rolling(selected_timeframe, center=True).mean()
+        / NEIGHBORHOOD_COUNCILS_NUMBER
+    )
+    neighborhood_sum_df["select_nc_ma"] = neighborhood_sum_df.counts.rolling(
+        selected_timeframe, center=True
+    ).mean()
 
-    neighborhood_sum_df = df[df.council_name == selected_council].groupby(['created_date']).agg('sum').reset_index()  # noqa
-    total_sum_df = df.groupby(['created_date']).agg('sum').reset_index()
-    total_sum_df["nc_avg"] = total_sum_df["counts"] / 99
-    merged_df = neighborhood_sum_df.merge(total_sum_df["nc_avg"].to_frame(), left_index=True, right_index=True)  # noqa
+    merged_df = pd.merge(
+        neighborhood_sum_df, total_sum_df, on=["created_date", "created_date"]
+    )
 
     fig = px.line(
         merged_df,
         x="created_date",
-        y=['counts', 'nc_avg'],
+        y=["select_nc_ma", "nc_ma"],
         color_discrete_sequence=DISCRETE_COLORS,
-        labels=LABELS,
-        title="Comparison trend for " + selected_council
+        labels={"created_date": "311 Request Date", "value": "Total Requests"},
+        title=selected_council + " vs 99 Neighborhood Councils 311 Requests Average",
```

Comment:
same: "Neighborhood Council Average"

Reply:
We should probably make a constant for this as well so that we don't need to worry about updating it in multiple places.

Reply:
what do you think we should make a constant for?

Reply:
gotcha